lib/v8.js

```diff
@@ -145,6 +146,8 @@ const {
   heapStatisticsBuffer,
   heapCodeStatisticsBuffer,
   heapSpaceStatisticsBuffer,
+  getCppHeapStatistics: _getCppHeapStatistics,
+  detailLevel,
```

Comment:
nit: Rather than exporting the `detailLevel` enum, I would just pass the string argument down and handle it at the c++ level. Alternatively, instead of using a string for the `type` argument below, use an options like...

```
getCppHeapStatistics({ detailed: true });
getCppHeapStatistics({ detailed: false });
```

Reply:
Yes but this is quite unlikely to be a hot path so I'm not too concerned about performance and would rather simplify the js side of the code a bit more.

Reply:
If it's a string to the C++ land, then the C++ land needs to do the string to enum mapping instead of just static_cast (unless we want to clamp a V8 enum into a boolean, which could make future extensions difficult once it has more than 2 values..).

Reply:
> Yes but this is quite unlikely to be a hot path so I'm not too concerned about performance

If this is used by performance tracing then the overhead likely matters (or from https://github.com/nodejs/node/pull/4463l , why v8.getHeapStatistics() is optimized to be the way it is. It seems better to be consistent).

Reply:
Fair enough :-)